### PR TITLE
executor: split a separate region for index when split table region

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* Split a separate region for index when splitting table region.
* Split a separate region for the next index when splitting table index region.

**before**
```sql
create table t (a int,b int, index idxa(a), index idxb(b));
split table t between (0) and (10000) regions 2;

-- The key/value of index `idxa` and `idxb` will write into region 251. But User will think region 251 and region 3 are only used to store record kv.
show table t regions;
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+
| REGION_ID | START_KEY   | END_Key     | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING |
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+
| 251       | t_52_       | t_52_r_5000 | 254       | 6               | 254, 255, 256 | 0          |
| 3         | t_52_r_5000 |             | 134       | 6               | 4, 16, 134    | 0          |
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+

split table t index idxa between (0) and (10000) regions 2;

-- The key/value of `idxb` and some part of index `idxa` will write into region 251.
show table t regions;
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+
| REGION_ID | START_KEY                               | END_Key                                 | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING |
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+
| 251       | t_52_i_1_038000000000001388030000000000 | t_52_r_5000                             | 254       | 6               | 254, 255, 256 | 0          |
| 3         | t_52_r_5000                             |                                         | 134       | 6               | 4, 16, 134    | 0          |
| 262       | t_52_i_1_                               | t_52_i_1_038000000000001388030000000000 | 266       | 1               | 264, 265, 266 | 0          |
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+
```

In This PR
```sql
create table t (a int,b int, index idxa(a), index idxb(b));
split table t between (0) and (10000) regions 2;

-- The key/value of index `idxa` and `idxb` will write into region 271. Region 275 and region 3 is only used to store record key/values/
show table t regions;
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+
| REGION_ID | START_KEY   | END_KEY     | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING |
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+
| 275       | t_55_r      | t_55_r_5000 | 278       | 6               | 278, 279, 280 | 0          |
| 3         | t_55_r_5000 |             | 134       | 6               | 4, 16, 134    | 0          |
| 271       | t_55_       | t_55_r      | 274       | 6               | 272, 273, 274 | 0          |
+-----------+-------------+-------------+-----------+-----------------+---------------+------------+

split table t index idxa between (0) and (10000) regions 2;

-- The key/value of index `idxb` will write into region 271, region 291 and region 286 will only used to store index `idxa` key/values.
show table t regions;
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+
| REGION_ID | START_KEY                               | END_KEY                                 | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING |
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+
| 275       | t_55_r                                  | t_55_r_5000                             | 278       | 6               | 278, 279, 280 | 0          |
| 3         | t_55_r_5000                             |                                         | 16        | 2               | 4, 16, 134    | 0          |
| 291       | t_55_i_1_                               | t_55_i_1_038000000000001388030000000000 | 296       | 7               | 292, 295, 296 | 0          |
| 286       | t_55_i_1_038000000000001388030000000000 | t_55_i_2_                               | 289       | 6               | 288, 289, 290 | 0          |
| 271       | t_55_i_2_                               | t_55_r                                  | 274       | 6               | 272, 273, 274 | 0          |
+-----------+-----------------------------------------+-----------------------------------------+-----------+-----------------+---------------+------------+

```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -4089,28 +4089,26 @@ func (s *testSuiteP1) TestSplitRegion(c *C) {
...
+	tk.MustExec("drop table if exists t_regions")
+	tk.MustExec("create table t_regions (a int key, b int, c int, index idx(b), index idx2(c))")
 
 	// Test show table regions.
-	tk.MustExec(`split table t_regions1 by (0)`)
-	tk.MustQuery(`split table t_regions between (-10000) and (10000) regions 4;`).Check(testkit.Rows("3 1"))
+	tk.MustQuery(`split table t_regions between (-10000) and (10000) regions 4;`).Check(testkit.Rows("4 1"))
 	re := tk.MustQuery("show table t_regions regions")
 	rows := re.Rows()
 	// Table t_regions should have 4 regions now.
```
Comment (AilinKid) [inline executor/executor_test.go]: should change the comment now.

Comment (crazycs520) [reply executor/executor_test.go]: Great catch. done.

**Pull request merged**